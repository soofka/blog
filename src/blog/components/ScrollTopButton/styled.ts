import styledComponents from 'styled-components';

export const StyledScrollTopButton = styledComponents.a`
  float: left;
  width: 100%;
  margin: ${(props) => props.theme.marginM} 0;
`;
