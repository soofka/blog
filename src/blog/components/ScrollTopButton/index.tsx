import * as React from 'react';

import { scrollToTop } from 'blog/common/helpers';
import { Label } from 'blog/components/Label';

import { StyledScrollTopButton } from './styled';

export const ScrollTopButton = (): JSX.Element => {
  return(
    <StyledScrollTopButton
      href="#"
      onClick={(e) => {
        e.preventDefault();
        scrollToTop();
      }}
    >
      &uarr; <Label name="scroll_to_top"/>
    </StyledScrollTopButton>
  );
};
