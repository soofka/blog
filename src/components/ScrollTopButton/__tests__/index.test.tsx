import * as React from 'react';
import { shallow } from 'enzyme';

import ScrollTopButton from 'components/ScrollTopButton';

describe('<ScrollTopButton/>', () => {

  it('renders correctly', () => {
    expect(shallow(<ScrollTopButton/>)).toMatchSnapshot();
  });

});
