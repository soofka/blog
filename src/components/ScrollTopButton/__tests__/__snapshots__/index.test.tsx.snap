// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollTopButton/> renders correctly 1`] = `
<styled.a
  href="#"
  onClick={[Function]}
>
  â†‘ 
  <inject-_class-with-languageStore
    name="scroll_to_top"
  />
</styled.a>
`;
